### PR TITLE
Task02: Practice bash

### PR DESCRIPTION
**Pull request by deesync**
- Made a backup script. The script is compressing source code with gzip and copying files to chosen release folder
- Also created a script that starts the game from task1 in an infinite loop with the prompt to ask user for exit or continue. 
The script uses game return codes to count successful and failed attempts. 
To improve the readability of the output, the script uses colored messages

### CODE REVIEW

File: task2-bash/backup.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -ex
+
+src_folder="../task1-simple-program"
+tmp_folder="tmp/guesanumber"
+
+
+## Copying files to tmp folder
+mkdir -p $tmp_folder
+cp $src_folder/*.{c,h} $tmp_folder
```
Comment (hrandib) [inline task2-bash/backup.sh]: Paths in quotes is a good practice

File: task2-bash/backup.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+## Archiving backup to release folder
+mkdir -p $src_folder/release
+tar --directory=$(dirname $tmp_folder) -cvzf \
+	$src_folder/release/$(basename $tmp_folder).tar.gz \
+	$(basename $tmp_folder)
+
```
Comment (hrandib) [inline task2-bash/backup.sh]: Please don't forget to add this newline in all files

File: task2-bash/backup.sh [hunk 3]
```diff
@@ -1,9 +1,12 @@
 #!/bin/bash -ex
 
 src_folder="../task1-simple-program"
-tmp_folder="tmp/guesanumber"
+tmp_folder="/tmp/guesanumber"
 
 
+## Flushing tmp directory
+[ -d $tmp_folder ] && rm $tmp_folder/*
```
Comment (hrandib) [inline task2-bash/backup.sh]: rm -r may neccessary

File: task2-bash/backup.sh [hunk 4]
```diff
@@ -1,19 +1,19 @@
 #!/bin/bash
```
Comment (hrandib) [inline task2-bash/backup.sh:1]: As for me, the previous version was fine. You've truncated some useful information

File: task2-bash/backup.sh [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
 
-src_folder="../task1-simple-program"
-tmp_folder="/tmp/guesanumber"
-release_folder="$src_folder/release"
+src="../task1-simple-program"
+tmp="/tmp/guesanumber"
+release="$src/release"
```
Comment (hrandib) [inline task2-bash/backup.sh:5]: Release version? Release date?

File: task2-bash/guess.sh [hunk 6]
```diff
@@ -34,7 +34,7 @@ run_game() {
 		((total_tries++))
 	else
 		echo "Something goes wrong..."
-		exit 2;
+		exit 10;
```
Comment (hrandib) [inline task2-bash/guess.sh:37]: It's a good practice to define constants for the exit codes for scripts non-oneliner scripts

**Review by hrandib — CHANGES_REQUESTED** [review]
Please resolve all the comments

File: task2-bash/backup.sh (hunk 1 above)
Comment (deesync) [reply task2-bash/backup.sh]: Fixed https://github.com/Kernel-GL-HRK/gl_kernel_BaseCamp_2022_II/pull/33/commits/7db8264394124e8030fcdbdcfa961a1690d659c1

File: task2-bash/backup.sh (hunk 2 above)
Comment (deesync) [reply task2-bash/backup.sh]: Yes. This has already been fixed in subsequent commits

File: task2-bash/backup.sh (hunk 3 above)
Comment (deesync) [reply task2-bash/backup.sh]: I didn't use -r key here intentionally to affect only files

File: task2-bash/guess.sh (hunk 6 above)
Comment (deesync) [reply task2-bash/guess.sh]: Fixed https://github.com/Kernel-GL-HRK/gl_kernel_BaseCamp_2022_II/pull/33/commits/265aead3cd1e5b3a4d26ad359b734fcddd1a1d6f

**Review by hrandib — APPROVED** [review]

**Pull request merged**